File: linked lists/82. Remove Duplicates from Sorted List II/deleteDuplicates.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        curr = head
        prev = None
        
        while curr and curr.next:
            if curr.val == curr.next.val:
                tmp = curr
                v = tmp.val

                # While there are duplicates
                while tmp and tmp.val == v:
                    tmp = tmp.next

                if prev:
                    prev.next = tmp
                else:
                    # If there's no previous node, then we are at the beginning
                    head = tmp

                curr = tmp
            else:
                prev = curr
                curr = curr.next
        
        return head

File: linked lists/82. Remove Duplicates from Sorted List II/deleteDuplicates.js
class ListNode {
    constructor(val = 0, next = null) {
      this.val = val;
      this.next = next;
    }
  }
  
  var deleteDuplicates = function(head) {
    let curr = head;
    let prev = null;
  
    while (curr && curr.next) {
      if (curr.val === curr.next.val) {
        let tmp = curr;
        let v = tmp.val;
  
        // Skip duplicates
        while (tmp && tmp.val === v) {
          tmp = tmp.next;
        }
  
        if (prev) {
          prev.next = tmp;
        } else {
          head = tmp;
        }
  
        curr = tmp;
      } else {
        prev = curr;
        curr = curr.next;
      }
    }
  
    return head;
  };
  